lib/Graph/Grad.cpp

```diff
@@ -257,6 +257,23 @@ Function *glow::differentiate(Function *F, const TrainingConfig &conf,
       continue;
     }
 
+    if (N->getKind() == Kind::BatchedReduceAddNodeKind) {
+      BatchedReduceAddNode *BRA = cast<BatchedReduceAddNode>(N);
+      // Get gradient.
+      NodeValue OutputG = map.getGradient(Result);
+      // Get input value.
+      NodeValue Input = BRA->getBatch();
+
+      // Gradient for BatchedReduceAddNode is TileNode,
+      // repeating OutputG batch times.
+      auto Num = Input.dims()[0] / OutputG.dims()[0];
+      auto *TN = new TileNode("tile.grad", Input.getType(), OutputG, Num, 0);
```

Comment:
Consider adding a comment to explain at the last argument is the tiling axis:
`auto *TN = new TileNode("tile.grad", Input.getType(), OutputG, Num, /*axis=*/0);`